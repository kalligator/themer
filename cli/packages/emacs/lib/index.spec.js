const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer emacs theme generator', () => {
  const promisedFiles = Promise.all(render(colors));

  it('should produce valid theme files', async () => {
    const files = await promisedFiles;
    expect(files.length).toBe(2);
    files.forEach((file) => {
      const contents = file.contents.toString('utf8');
      expect(/undefined/.test(contents)).toBe(false);
    });
  });

  it('should provide installation instructions', async () => {
    const files = await promisedFiles;
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
