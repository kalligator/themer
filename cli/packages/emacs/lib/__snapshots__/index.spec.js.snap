// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer emacs theme generator should provide installation instructions 1`] = `
"
Copy (or symlink) the generated theme files into \`~/.emacs.d/\`:

    cp 'themer-dark-theme.el' ~/.emacs.d/
    cp 'themer-light-theme.el' ~/.emacs.d/

Then load the desired theme in \`init.el\`:

    (load-theme 'themer-dark t)
    ;; or
    (load-theme 'themer-light t)
"
`;
